httpclient5/src/main/java/org/apache/hc/client5/http/protocol/NextNonceInterceptor.java

```diff
@@ -0,0 +1,146 @@
+ * utility class for robust token parsing.</p>
+ *
+ * <p>In the context of HTTP Digest Access Authentication, the {@code nextnonce} parameter is
+ * a critical part of the security mechanism, designed to mitigate replay attacks and enhance mutual
+ * authentication security. It provides the server with the ability to set and enforce single-use or
+ * session-bound nonces, prompting the client to use the provided {@code nextnonce} in its next request.
+ * This setup helps secure communication by forcing new cryptographic material in each transaction.
+ * </p>
+ *
+ * <p>This interceptor is stateless and thread-safe, making it suitable for use across multiple
+ * threads and HTTP requests. It should be registered with the HTTP client to enable support
+ * for advanced authentication mechanisms that require tracking of nonce values.</p>
+ *
+ * @since 5.5
+ */
+
+@Contract(threading = ThreadingBehavior.STATELESS)
+public class NextNonceInterceptor implements HttpResponseInterceptor {
+
+    public static final HttpResponseInterceptor INSTANCE = new NextNonceInterceptor();
+
+    private final Tokenizer tokenParser = Tokenizer.INSTANCE;
+
+
+    private static final String AUTHENTICATION_INFO_HEADER = "Authentication-Info";
+
+    private static final Tokenizer.Delimiter TOKEN_DELIMS = Tokenizer.delimiters('=', ',');
+    private static final Tokenizer.Delimiter VALUE_DELIMS = Tokenizer.delimiters(',');
+
+    /**
+     * Processes the HTTP response and extracts the {@code nextnonce} parameter from the
+     * {@code Authentication-Info} header if available, storing it in the provided {@code context}.
+     *
+     * @param response the HTTP response containing the {@code Authentication-Info} header
+     * @param entity   the response entity, ignored by this interceptor
+     * @param context  the HTTP context in which to store the {@code nextnonce} parameter
+     * @throws NullPointerException if either {@code response} or {@code context} is null
+     */
+    @Override
+    public void process(final HttpResponse response, final EntityDetails entity, final HttpContext context) {
+        Args.notNull(response, "HTTP response");
+        Args.notNull(context, "HTTP context");
+
+        final Header header = response.getFirstHeader(AUTHENTICATION_INFO_HEADER);
+        if (header != null) {
+            final String nextNonce;
+            if (header instanceof FormattedHeader) {
+                final CharArrayBuffer buf = ((FormattedHeader) header).getBuffer();
+                final ParserCursor cursor = new ParserCursor(0, buf.length());
+                cursor.updatePos(((FormattedHeader) header).getValuePos());
+                nextNonce = parseNextNonce(buf, cursor);
+            } else {
+                final String headerValue = header.getValue();
+                final ParserCursor cursor = new ParserCursor(0, headerValue.length());
+                final CharArrayBuffer buf = new CharArrayBuffer(headerValue.length());
```

Comment:
@arturobernalg This intermediate buffer will become unnecessary.

Reply:
done.